mllib/src/main/scala/org/apache/spark/ml/regression/GeneralizedLinearRegression.scala

```diff
@@ -1441,4 +1460,33 @@ class GeneralizedLinearRegressionTrainingSummary private[regression] (
         "No p-value available for this GeneralizedLinearRegressionModel")
     }
   }
+
+  /**
+   * Summary table with feature name, coefficient, standard error,
+   * tValue and pValue.
+   */
+  @Since("2.2.0")
+  lazy val summaryTable: DataFrame = {
+    if (isNormalSolver) {
+      var featureNamesLocal = featureNames
+      var coefficients = model.coefficients.toArray
+      var idx = Array.range(0, coefficients.length)
+      if (model.getFitIntercept) {
+        featureNamesLocal = featureNamesLocal :+ Intercept
+        coefficients = coefficients :+ model.intercept
+        // Reorder so that intercept comes first
+        idx = (coefficients.length - 1) +: idx
+      }
+      val result = for (i <- idx.toSeq) yield
+        (featureNamesLocal(i), coefficients(i), coefficientStandardErrors(i),
+        tValues(i), pValues(i))
+
+      val spark = SparkSession.builder().getOrCreate()
+      import spark.implicits._
+      result.toDF("Feature", "Coefficient", "StdError", "TValue", "PValue").repartition(1)
```

Comment:
Could you let me know the reason of wrapping the result as a ```DataFrame```? I think a local 2D array is enough. ```DataFrame``` adds some extra cost and actually it will collect as local array when you call ```toString```.